src/jsc/jsc_class.hpp

```diff
@@ -23,9 +23,47 @@
 #include "js_class.hpp"
 #include "js_util.hpp"
 
+namespace realm {
+	namespace js {
+		template<typename T>
+		struct RealmObjectClass;
+	}
+	namespace node {
+		struct Types;
+	}
+}
+
 namespace realm {
 namespace jsc {
 
+static js::Protected<JSObjectRef> ObjectDefineProperty;
+static js::Protected<JSObjectRef> FunctionPrototype;
+static js::Protected<JSObjectRef> RealmObjectClassConstructor;
+static js::Protected<JSObjectRef> RealmObjectClassConstructorPrototype;
+
+static inline void jsc_class_init(JSContextRef ctx, JSObjectRef globalObject) {
+    //handle ReactNative app refresh by reseting the cached constructor values
+    if (RealmObjectClassConstructor) {
+        RealmObjectClassConstructor = js::Protected<JSObjectRef>();
+    }
+
+    if (RealmObjectClassConstructorPrototype) {
+        RealmObjectClassConstructorPrototype = js::Protected<JSObjectRef>();
+    }
+
+    JSValueRef value = nullptr;
+    value = jsc::Object::get_property(ctx, globalObject, "Object");
+    JSObjectRef objectClass = jsc::Value::to_object(ctx, value);
+
+    value = jsc::Object::get_property(ctx, objectClass, "defineProperty");
+    ObjectDefineProperty = js::Protected<JSObjectRef>(ctx, Value::to_object(ctx, value));
+
+    value = jsc::Object::get_property(ctx, globalObject, "Function");
+    JSObjectRef globalFunction = jsc::Value::to_object(ctx, value);
+    value = jsc::Object::get_property(ctx, globalFunction, "prototype");
+    FunctionPrototype = js::Protected<JSObjectRef>(ctx, Value::to_object(ctx, value));
```

Comment:
Probably just a matter of personal preference, but when you reuse `JSValueRef value` here, the ordering of lines becomes more significant, making it more vulnerable to bugs if the code is reordered / refactored.

What's the downside to having individual values or passing the result of `get_property` directly to `to_object`?

Reply:
`value` is intermediate variable which is used just to follow the rule "no two statements with side effects on single line". 
Reusing the variable could not possibly lead to bugs if the one doing a refactor follows the rule that value is intermediate variable and is used only for that. The same goes for JSValueRef exception statements reusing that are present on certain places.
Reusing this is for code clarity and readability.

Reply:
`jsc::Object::get_property` is a pure function, isn't it?

Reply:
It returns an intermediate value which can be null pointer which is hard to debug if the code is not structured with that in mind. I don't see variable reuse as an issue. In fact it allows for more readable code. Thanks for the suggestion but I would like to move on with this.